src/strands/session/session_manager.py

```diff
@@ -20,19 +28,40 @@ class SessionManager(HookProvider, ABC):
     for an agent, and should be persisted in the session.
     """
 
-    def register_hooks(self, registry: HookRegistry, **kwargs: Any) -> None:
-        """Register hooks for persisting the agent to the session."""
-        # After the normal Agent initialization behavior, call the session initialize function to restore the agent
-        registry.add_callback(AgentInitializedEvent, lambda event: self.initialize(event.agent))
-
-        # For each message appended to the Agents messages, store that message in the session
-        registry.add_callback(MessageAddedEvent, lambda event: self.append_message(event.message, event.agent))
+    def __init__(self, session_type: SessionType = SessionType.AGENT) -> None:
+        """Initialize SessionManager with session type.
 
-        # Sync the agent into the session for each message in case the agent state was updated
-        registry.add_callback(MessageAddedEvent, lambda event: self.sync_agent(event.agent))
+        Args:
+            session_type: Type of session (AGENT or MULTI_AGENT)
+        """
+        self.session_type: SessionType = session_type
```

Comment:
Is there a need to have a session-manager work for only Agents OR MultiAgent?  Can it just work for both by subscribing to all events?

Apologies if this was already hashed out with @Unshure - if it was, can you summarize the discussion in the PR summary